packages/database/scripts/setupNewDatabase.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash -e
+
+SCRIPT_DIR=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
+cd "$SCRIPT_DIR"
+cd ..
+
+if [ "$1" != "OK" ]; then
+  echo "Note: this script reads .env vars"
+  echo "   [1] Update DB_NAME etc... in database db.env"
+  echo ""
+  echo "re-run with OK to continue"
+  exit 1;
+fi
+
+DIR=$(pwd "$0")
+source "$DIR/../../scripts/bash/mergeEnvForDB.sh" 
+
+# Set default port in case it wasn't in .env
+: "${DB_PORT:=5432}"
+
+set -x
+
+IS_RDS=$(PGPASSWORD=$DB_PG_PASSWORD psql -qtAX -h $DB_URL -p $DB_PORT -U $DB_PG_USER -c "SELECT 1 FROM pg_roles WHERE rolname='rds_superuser'")
+
+PGPASSWORD=$DB_PG_PASSWORD psql -h $DB_URL -p $DB_PORT -U $DB_PG_USER -c "CREATE ROLE $DB_USER LOGIN PASSWORD '$DB_PASSWORD'" || echo "Role $DB_USER already exists?"
+if [ "$IS_RDS" = "1" ]; then
+  PGPASSWORD=$DB_PG_PASSWORD psql -h $DB_URL -p $DB_PORT -U $DB_PG_USER -c "GRANT rds_superuser TO $DB_USER"
+  PGPASSWORD=$DB_PG_PASSWORD psql -h $DB_URL -p $DB_PORT -U $DB_PG_USER -c "grant $DB_USER to postgres"
+else
+  PGPASSWORD=$DB_PG_PASSWORD psql -h $DB_URL -p $DB_PORT -U $DB_PG_USER -c "ALTER USER $DB_USER WITH SUPERUSER"
+fi
+PGPASSWORD=$DB_PG_PASSWORD psql -h $DB_URL -p $DB_PORT -U $DB_PG_USER -c "CREATE ROLE tupaia_read PASSWORD '$DB_PASSWORD'" || echo "Role tupaia_read already exists?"
+PGPASSWORD=$DB_PG_PASSWORD psql -h $DB_URL -p $DB_PORT -U $DB_PG_USER -c "CREATE DATABASE $DB_NAME WITH OWNER $DB_USER"
+PGPASSWORD=$DB_PG_PASSWORD psql -h $DB_URL -p $DB_PORT -U $DB_PG_USER $DB_NAME -c "CREATE EXTENSION IF NOT EXISTS postgis"
+PGPASSWORD=$DB_PASSWORD psql -h $DB_URL -p $DB_PORT -U $DB_USER -d $DB_NAME -f schema/schema.sql
+
+if [ "$IS_RDS" = "1" ]; then
+  PGPASSWORD=$DB_PG_PASSWORD psql -h $DB_URL -p $DB_PORT -U $DB_PG_USER -c "REVOKE rds_superuser FROM $DB_USER"
+else
+  PGPASSWORD=$DB_PG_PASSWORD psql -h $DB_URL -p $DB_PORT -U $DB_PG_USER -c "ALTER USER $DB_USER WITH NOSUPERUSER"
+fi
+
+set +x
+
+echo "Installing mvrefresh"
+DB_NAME=$DB_NAME yarn workspace @tupaia/data-api install-mv-refresh
+echo "Patching mvrefresh"
+DB_NAME=$DB_NAME yarn workspace @tupaia/data-api patch-mv-refresh up
+echo "Installing Analytics table"
+DB_NAME=$DB_NAME yarn workspace @tupaia/data-api build-analytics-table
+
+echo "Deleting migrations that target data modifications, as there is no data to migrate in the new database"
+rm -rf ./src/migrations-backup
+mkdir  ./src/migrations-backup
+cp -r ./src/migrations/* ./src/migrations-backup/
+rm ./src/migrations/*modifies-data.js
+DB_NAME=$DB_NAME yarn migrate
+cp -r ./src/migrations-backup/* ./src/migrations/
```

Comment:
Why do we restore the data migrations? It means they'll be caught as not run next update, but shouldn't need to run there, right?

Reply:
Hmmmm that's quite a good point...

This script is used to effectively setup the schema of the database, and it's used in contexts other than a new deployment (really just setting up the test database).

I think the issue with data migrations is a separate one, and one we haven't really got a solution for at the moment. We could delete them manually, but then they'd be sitting there as a git diff forever which doesn't feel right.

This PR doesn't quite take us all the way to being ready to deploy a new version of tupaia on top of the LESMIS database (I decided to draw the line at just getting the scripts to spin up an instance). I'll link this convo in the ticket where we address upgrading over the LESMIS database. Happy to leave this as is for now then?

Reply:
Got it, yeah that's cool to bring forward as a future thing